core/src/main/scala/org/http4s/Uri.scala

```diff
@@ -319,6 +318,105 @@ object Uri extends UriFunctions {
       }
   }
 
+  sealed abstract class Host {
+    def inetAddress: Either[UnknownHostException, InetAddress]
+    def value: String
+    def render(w: Writer): w.type
+
+    final override def equals(o: Any) = o match {
+      case that: Host => this.value.equalsIgnoreCase(that.value)
+      case _ => false
+    }
+
+    final override def hashCode(): Int = value.##
+
+    override final def toString = s"Host($value)"
+  }
+
+  object Host {
+    private def fromName(name: String) = new Host {
+      def inetAddress =
+        try Right(InetAddress.getByName(name))
```

Comment:
This can block a thread, which would be sad.  I don't know how best to discourage calling this.